### ISSUE TITLE
Add profile field position to Author Boxes Profile Fields

### ISSUE THREAD

**Issue by ojopaul** [publishpress/PublishPress-Authors#933]
@stevejburge 
I think we can allow profile field position re-order via drag and drop or button so elements can be in user desired order and not as in default list (e.g: First name first, Last name etc). 

This is something that i realised will come in handy while working on author boxes editor profile integration but it's not straight forward due to the whole metabox been sortable and inside drag and drop elements already and as a result, we need to devise a custom solution but due to limited time we have on processing current released, it may move to the next milestone.

In summary, this will enable user change author boxes profile fields element position.

In the long run, maybe we could consider same update for the whole design(drag and drop to change element position)

![profile-fields](https://user-images.githubusercontent.com/92122385/195145697-0f02b132-7f6d-4e28-a1fc-f44103f48feb.jpg)

**Comment by ojopaul** [comment]
@stevejburge Perhaps we should go for this once and for all and implement the solution for the design instead of limiting it to only profile?

Probably allowing drag and drop for the editor tab to change the tab element position and order?

Of course, this will be next milestone but i believe it'll be useful.

**Comment by stevejburge** [comment]
Thanks @ojopaul 

So we still create fields from the current screen, but arrange them on this new screen? That makes sense.

**Comment by ojopaul** [comment]
@stevejburge No, the screenshot above is for profile fields tab alone as i decided to add expand/collapse feature to the profile field tabs content due to it been too long and considering the fact pro users may have many profile fields.

But if we decided to roll out drag and drop for the whole layout, literally, the  designer will be able to change content order by dragging the items in below screenshot to change their order.

So, it's same screen and all highlighted element will be draggable to change position.


![drag](https://user-images.githubusercontent.com/92122385/195153600-b664e679-bb19-4bbf-801a-676d877f32b3.jpg)

**Comment by stevejburge** [comment]
Thanks @ojopaul 

Drag-and-drop for the whole Preview would be great but that sounds like a lot more work?

**Comment by ojopaul** [comment]
@stevejburge Adding the re-order features to the newly added profile fields tab items will fit in next release without too many work but doing so for all the UI will require more time.

So, we can implement for profile fields next milestone and move whole UI drag re-order feature to future milestone when we feel its neccessary.

**Comment by ojopaul** [comment]
This is taking more time than expected.

So, I'm removing it from this milestone till we have enough time allocated for it.

**Comment by ojopaul** [comment]
@stevejburge Do you mean ability to select each fields position?

That make sense and will work.

**Comment by stevejburge** [comment]
Yes, each field can be placed in the same row as the Title, or the Bio, or the Meta, etc...

It might be easier than building a drag-and-drop solution
